### ISSUE TITLE
GitLabApi class docs authorization token terminology does not match Gitlab docs

### ISSUE THREAD

**Issue by dwlocks** [gitlab4j/gitlab4j-api#503]
It seems that all the tokens referenced in the javadocs related to authentication have different names than the ones mentioned in the documentation for using gitlab proper.

For example, the gitlab documentation talks about "Personal Access Tokens"
https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#personal-access-tokens

Gitlab documentation talks about OAuth2.0 flows using terms like "APP_ID" "APP_SECRET" and "access_token".  The Gitlab UI is consistent in terminology with the Gitlab documentation.

JavaDocs for GitLabApi never mention these terms.  The JavaDocs use terms like username, password and "secret_token". The docs mention that we will get a "private token" which is similar to but still different from "personal access token".

I suspect don't understand some key concept.

How can I use any of the things I have (Application ID, Application Secret, Personal Access Token) to authenticate using the GitLabApi jar?

**Comment by dwlocks** [comment]
The README.md does mention the Personal Access Token, so there's a bright spot. This leads me to:

Gitlab Documentation Personal Access Token = JavaDoc privateToken.  

But there's still no way obvious way to translate the java code's terminology of "authToken" , "privateToken", and "secretToken" to the gitlab documentation.

An aside: README.md also says you can use the GitLabApi.login() methods, which the JavaDocs say are deprecated.

**Comment by gmessner** [comment]
GitLab renamed "Private Token" to "Personal Access Token" which is why you are seeing this inconsistency in the code, I suppose we could rename the parameters to eliminate the confusion.  That being said, when passing the "Personal Access Token" to the GitLab REST API it is passed in a header called ```PRIVATE-TOKEN``` or as a query param called ```private-token```, see this link for more info:  https://docs.gitlab.com/ee/api/README.html#personal-access-tokens

I believe GitLab made a mistake by renaming "Private Token" as "Personal Access Token" in the UI because "Personal Access Token" is too close in name to "Access Token" used by OAuth2.0.

As far as the Oauth2.0 flows, I believe you are confusing using GitLab as an Outh2.0 provider for external applications with using the ```oauth2Login()``` methods in the GitLabApi class for username/passwod authentication with the GitLab REST API.   

The deprecated ```login()``` methods are deprecated because GitLab deprecated and removed the ```Session``` endpoint that was used for authenticating with username and password, this library replaced that with the ```oath2Login()``` methods for username/password authentication with the GitLab REST API. 

The "secretToken" is something that you can provide to the ```oauth2Login()``` that the GitLab server will simply return in the ```X-Gitlab-Token``` HTTP header.  If you provide a "secretToken" and the value of the returned ```X-Gitlab-Token```  header does not match it,  the response will cause an exception to be thrown.

In the next release of GitLab4J-API, I will be deleting the deprecated ```login()``` methods and seeing what else I can do in the documentation to eliminate some of this confusion.  In the meantime you have 2 options for authenticating:

Personal Access Token use:
```
GitLabApi gitLabApi = new GitLabApi("http://your.gitlab.server.com", "YOUR_PERSONAL_ACCESS_TOKEN");
```
Username/Password use:
```
GitLabApi gitLabApi = GitLabApi.oauth2Login("http://your.gitlab.server.com", "your-username", "your-password");
```

**Comment by gmessner** [comment]
@dwlocks 
I have renamed the ```privateToken``` param ```personalAccessToken```, removed the deprecated ```login()``` methods, cleaned up the docs and have released version 4.14.15.   Please review and close this issue when you have time.

**Comment by dwlocks** [comment]
Thanks for the explanation. I was in fact confusing authenticating vs using gitlab as an OAuth2 provider.  (Also this is the fastest anyone has ever responded to one of my a github issues!)

The changes look good.  It would be nice if the gitlab documentation replaced "YOUR_ACCESS_TOKEN" with "PERSONAL_ACCESS_TOKEN" in https://docs.gitlab.com/ee/api/README.html#personal-access-tokens
